viewer/packages/tools/src/Measurement/MeasurementTool.ts

```diff
@@ -283,7 +287,10 @@ export class MeasurementTool extends Cognite3DViewerToolBase {
    * @param enable
    */
   setMeasurementLabelsVisible(enable: boolean): void {
-    this._htmlOverlay.visible(enable);
+    if (this._showMeasurements) {
+      this._htmlOverlay.visible(enable);
+      this._showLabels = enable;
+    }
```

Comment:
I still think we should be able to toggle the labels on and off "behind the scenes" even though the result will not be visible before the lines are made visible. If a user has hidden both labels and lines, and then want to show both simultaneously, it will be correct only if they turn on the lines first and then the labels, and not the other way around, the way this is written now. I think both orders should work, which they would if we keep track of whether the labels are toggled on regardless of whether they are visible or not. We can do this just by moving the `showLabels` line out of the if-block.